external/anomaly/tests/test_ote_anomaly_segmentation_training.py

```diff
@@ -0,0 +1,420 @@
+    return dataset, labels_schema
+
+def get_anomaly_segmentation_test_action_classes() -> List[Type[BaseOTETestAction]]:
+    return [
+        AnomalySegmentationTestTrainingAction,
+        OTETestTrainingEvaluationAction,
+        OTETestExportAction,
+        OTETestExportEvaluationAction,
+        OTETestPotAction,
+        OTETestPotEvaluationAction,
+        OTETestNNCFAction,
+        OTETestNNCFEvaluationAction,
+        OTETestNNCFExportAction,
+        OTETestNNCFExportEvaluationAction,
+        OTETestNNCFGraphAction,
+    ]
+
+class AnomalySegmentationTrainingTestParameters(DefaultOTETestCreationParametersInterface):
+    def test_case_class(self) -> Type[OTETestCaseInterface]:
+        return generate_ote_integration_test_case_class(
+            get_anomaly_segmentation_test_action_classes()
+        )
+
+    def test_bunches(self) -> List[Dict[str, Any]]:
+        # Extend with other datasets
+        test_bunches = [
+                dict(
+                    model_name=[
+                       'ote_anomaly_segmentation_padim',
+                       'ote_anomaly_segmentation_stfpm',
+                    ],
+                    dataset_name='mvtec_short',
+                    usecase='precommit',
+                ),
+                dict(
+                    model_name=[
+                       'ote_anomaly_segmentation_padim',
+                       'ote_anomaly_segmentation_stfpm',
+                    ],
+                    dataset_name='mvtec',
+                    patience=KEEP_CONFIG_FIELD_VALUE,
+                    batch_size=KEEP_CONFIG_FIELD_VALUE,
+                    usecase=REALLIFE_USECASE_CONSTANT,
+                ),
+        ]
+        return deepcopy(test_bunches)
+
+    def short_test_parameters_names_for_generating_id(self) -> OrderedDict:
+        DEFAULT_SHORT_TEST_PARAMETERS_NAMES_FOR_GENERATING_ID = OrderedDict(
+            [
+                ("test_stage", "ACTION"),
+                ("model_name", "model"),
+                ("dataset_name", "dataset"),
+                ("patience", "patience"),
+                ("batch_size", "batch"),
+                ("usecase", "usecase"),
+            ]
+        )
+        return deepcopy(DEFAULT_SHORT_TEST_PARAMETERS_NAMES_FOR_GENERATING_ID)
+
+    def test_parameters_defining_test_case_behavior(self) -> List[str]:
+        DEFAULT_TEST_PARAMETERS_DEFINING_IMPL_BEHAVIOR = [
+            "model_name",
+            "dataset_name",
+            "patience",
+            "batch_size",
+        ]
+        return deepcopy(DEFAULT_TEST_PARAMETERS_DEFINING_IMPL_BEHAVIOR)
+
+    def default_test_parameters(self) -> Dict[str, Any]:
+        DEFAULT_TEST_PARAMETERS = {
+            "patience": 1,
+            "batch_size": 2,
+        }
+        return deepcopy(DEFAULT_TEST_PARAMETERS)
+
+# TODO:sstrehlk: This function copies with minor change OTETestTrainingAction
+#             from ote_sdk.test_suite.
+#             Try to avoid copying of code.
```

Comment:
Is there a Jira ticket for this for reference? and for other todos in the code?

Reply:
I removed this comment, it was based on another part of code. I'don't know if Jira tickets were created for these todos.